File: app/src/main/java/com/zzzmode/appopsx/ui/model/PreAppInfo.java
package com.zzzmode.appopsx.ui.model;

import android.text.TextUtils;
import java.util.ArrayList;
import java.util.List;

/**
 * Created by zl on 2017/4/30.
 */

public class PreAppInfo {

  private String packageName;
  private String ignoredOps;
  private List<Integer> ops;

  public PreAppInfo(String packageName, String ignoredOps) {
    this.packageName = packageName;
    this.ignoredOps = ignoredOps;
  }

  public PreAppInfo(String packageName) {
    this.packageName = packageName;
  }

  public void setIgnoredOps(String ignoredOps) {
    this.ignoredOps = ignoredOps;
  }

  public String getPackageName() {
    return packageName;
  }

  public String getIgnoredOps() {
    return ignoredOps;
  }

  public List<Integer> getOps() {
    if (ops == null) {
      ops = new ArrayList<>();
      if (!TextUtils.isEmpty(ignoredOps)) {
        String[] split = ignoredOps.split(",");
        for (String s : split) {
          try {
            ops.add(Integer.valueOf(s));
          } catch (NumberFormatException e) {
            e.printStackTrace();
          }
        }
      }
    }
    return ops;
  }


  @Override
  public String toString() {
    return "PreAppInfo{" +
        "packageName='" + packageName + '\'' +
        ", ignoredOps='" + ignoredOps + '\'' +
        '}';
  }
}

File: app/src/main/kotlin/com/zzzmode/appopsx/ui/model/PreAppInfo.kt
package com.zzzmode.appopsx.ui.model

import android.text.TextUtils
import java.util.ArrayList

/**
 * Created by zl on 2017/4/30.
 */

class PreAppInfo {

    var packageName: String
        private set
    var ignoredOps: String? = null
    private var ops: MutableList<Int>? = null

    constructor(packageName: String, ignoredOps: String) {
        this.packageName = packageName
        this.ignoredOps = ignoredOps
    }

    constructor(packageName: String) {
        this.packageName = packageName
    }

    fun getOps(): List<Int> {
        if (ops == null) {
            ops = ArrayList()
            if (!TextUtils.isEmpty(ignoredOps)) {
                val split = ignoredOps!!.split(",".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
                for (s in split) {
                    try {
                        ops!!.add(Integer.valueOf(s))
                    } catch (e: NumberFormatException) {
                        e.printStackTrace()
                    }

                }
            }
        }
        return ops!!
    }


    override fun toString(): String {
        return "PreAppInfo{" +
                "packageName='" + packageName + '\''.toString() +
                ", ignoredOps='" + ignoredOps + '\''.toString() +
                '}'.toString()
    }
}
